pytorch_lightning/loops/optimizer/optimizer_loop.py

```diff
@@ -65,8 +65,7 @@ def connect(self, **kwargs: "Loop") -> None:
         raise NotImplementedError(f"{self.__class__.__name__} does not connect any child loops.")
 
     def reset(self) -> None:
-        if not self.restarting:
-            self.optim_progress.optimizer_idx = 0
+        self.optim_progress.optimizer_idx = 0
```

Comment:
reviewer: we don't actually have enough information to resume an optimizer loop here. so let's not pretend we can.

Reply:
the knowledge whether the idx needs to be reset to 0 or not. this depends on whether there was a failure or not. 
plus hidden state and all the output handling and also which optimizer currently is active

Reply:
keeping it, but have to add done() check.